### ISSUE TITLE
Setup GitHub Actions in order to avail permalink for each built version of openapi.yaml

### ISSUE THREAD

**Issue by ashutshkumr** [open-traffic-generator/models#5]
Follow the example for,
src: https://github.com/protocolbuffers/protobuf
build: https://github.com/protocolbuffers/protobuf/releases/tag/v3.13.0

@ajbalogh I'm assigning it to myself. Is it ok to,
- Use version in file name or put the version inside file itself ?
- Add Dockerfile to project itself

**Comment by ashutshkumr** [comment]
@ajbalogh I have this ready on a forked repository: https://github.com/ashutshkumr/openactions
Waiting for you to take a quick look so I can create a PR.

- Whenever user makes a change, they can decide if their change is worth designating a release number.
  If so, ```./do.sh version 1.0.1``` will,
  - update all YAML files with provided version
  - commit those changes, create a tag release and push
- https://github.com/ashutshkumr/openactions/blob/master/.github/workflows/main.yml detects a new tag and initiates github CI/CD action, which calls `./do.sh art` to:
  - create openapi.yaml bundle
  - use openapi.yaml to create openapi.json and protoapi.tar.gz spec
  - use openapi.yaml to create python client SDK
- As part of the workflow:
  - A new release is created with provided version. e.g. https://github.com/ashutshkumr/openactions/releases/tag/v1.0.8
  - Push all artifacts created in last step to the newly created release
  - Send an email

Status of all jobs can be checked here: https://github.com/ashutshkumr/openactions/actions

**Comment by ajbalogh** [comment]
That looks great - thanks for that!

**Comment by ajbalogh** [comment]
Is it possible for you to also add an email action for when the workflow is complete?

**Comment by ashutshkumr** [comment]
I just directly pushed all the changes (sorry about that, it's always better to open a PR) to check if it works. It does but unfortunately, I did not receive any email (maybe it has something to do with permission). e.g. I am receiving email on my private fork.

@ajbalogh , could you please check if you received any ? (https://github.com/open-traffic-generator/models/releases)

Another thing I see is not everything is visible (to me) on landing page - e.g. "Releases" section should be visible on RHS.
https://github.com/ashutshkumr/openactions vs https://github.com/open-traffic-generator/models

We might have to change some settings to show relevant information that may currently be hidden.

**Comment by ajbalogh** [comment]
got the notification and i enabled the releases to show on the landing page
of the repo

**Comment by ajbalogh** [comment]
i did the following to create a tag but did not see the version number in the generated release openapi.yaml change - am i missing something?
git tag -v0.0.1 -m "annotation message here"
git push origin v0.0.1

**Comment by ashutshkumr** [comment]
We could also proceed like this:
If somebody pushes a new tag, update version numbers in openapi yaml files. But we will end up with inconsistency where versions inside yaml files are upgraded after tag release.

**Comment by ajbalogh** [comment]
i think it should be part of the tag process and the only time it can be updated - as opposed to be user initiated from within the docker.
so regardless of what is done to the files they will be normalized as part of the tag process.

what do you think?
________________________________

**Comment by ashutshkumr** [comment]
Definitely, I'll look into it tomorrow. Will push 0.0.2 for now (with proper version updates).

**Comment by ajbalogh** [comment]
excellent - thanks!!!
________________________________

**Comment by ashutshkumr** [comment]
So https://github.com/open-traffic-generator/models/commit/2385c0acb73081ba15f87bffd1e23bf4a3bb8e40 got a little complicated.

To summarize,
- Now we have per-push workflow, so we may delete tavis ci yaml 
- To create new release, just update the version in `api/api.yaml` and push (no need to create/push tags)

More details with comments: https://github.com/open-traffic-generator/models/blob/master/.github/workflows/workflow.yml 

I'll explain more here based on intended `goals`:

- Perform `Actions` and generate artifacts on every push - Done.
  But the artifacts are posted to `Actions` page and are not permanent (persists for about 90 days).
  *NOTE*: Creating a tag and release for every push is bad practice and hence I refrained.

- Create Release only when intended, and it should reflect in base and generated .yaml - Done.
  As part of the workflow, if version number is changed in `api/api.yaml` (and if there are no tags for that version number), then create new tag + release and upload artifacts to that release page as well (persists unless release page is deleted)

- Avoid commit pollution. Do not make changes to all yaml files every time there's a new release - Done.
  As mentioned above, new releases are now only created upon detecting change in `api/api.yaml`. Thus other yaml files need not be updated. But, to be semantically correct, I have versioned all those files as `^0.0.0` which is ignored by bundler.py anyway.

**Comment by ashutshkumr** [comment]
So https://github.com/open-traffic-generator/models/commit/2385c0acb73081ba15f87bffd1e23bf4a3bb8e40 got a little complicated.

To summarize,
- Now we have per-push workflow, so we may delete tavis ci yaml 
- To create new release, just update the version in `api/api.yaml` and push (no need to create/push tags)

More details with comments: https://github.com/open-traffic-generator/models/blob/master/.github/workflows/workflow.yml 

I'll explain more here based on intended `goals`:

- Perform `Actions` and generate artifacts on every push - Done.
  But the artifacts are posted to `Actions` page and are not permanent (persists for about 90 days).
  *NOTE*: Creating a tag and release for every push is bad practice and hence I refrained.

- Create Release only when intended, and it should reflect in base and generated .yaml - Done.
  As part of the workflow, if version number is changed in `api/api.yaml` (and if there are no tags for that version number), then create new tag + release and upload artifacts to that release page as well (persists unless release page is deleted)

- Avoid commit pollution. Do not make changes to all yaml files every time there's a new release - Done.
  As mentioned above, new releases are now only created upon detecting change in `api/api.yaml`. Thus other yaml files need not be updated. But, to be semantically correct, I have versioned all those files as `^0.0.0` which is ignored by bundler.py anyway.
